docs/design/README.md

```diff
@@ -0,0 +1,595 @@
+
+### Background
+
+One of Bottlerocket's security features is its immutable root filesystem.
+Because software cannot be installed on Bottlerocket after the image has been created, it lacks a traditional package manager.
+Variants allow for different software and settings to be defined in Bottlerocket at build time.
+This improves security by ensuring that only the minimum software is installed to support a specific use case.
+
+Currently, in order to define a variant, it needs to be added to the `variants` directory of the Bottlerocket git repository.
+We call this *in-tree*, meaning that a variant can only be defined within Bottlerocket's git tree.
+Similarly, to add software to Bottlerocket, you need add it to the in-tree `packages` directory.
+
+The fundamental goal of variants is to support customizing Bottlerocket for different use cases.
+However, doing so currently requires forking and altering the Bottlerocket git repo.
+To stay up-to-date with core project would require frequent rebasing with high potential for conflicting changes in the core project.
+
+This document describes `twoliter`, a tool that will allow for the definition of a Bottlerocket variant externally from the main Bottlerocket project's git tree.
+Eventually `twoliter` will interface with another project underway to allow out-of-tree builds to define their own settings.
+Integration with the new settings system is out of scope for this document, and will be integrated with settings as both projects mature.
+
+### Glossary
+
+Here are a few of the terms we will use in this document:
+
+- **In-Tree**: Anything that exists in, or is defined by, the [main Bottlerocket git repository]
+- **Out-of-Tree Build (OOTB)**: User-customized builds of Bottlerocket defined by an external git repository owned by the user.
+- **Out-of-Tree Variants**: Synonymous with *Out of Tree Builds*.
+- **Maintainer**: A person who creates Bottlerocket variants using Twoliter.
+- **Kit**: A container image with RPM packages in it.
+
+[main Bottlerocket git repository]: https://github.com/bottlerocket-os/bottlerocket
+
+### Requirements
+
+The `twoliter` build tool should meet the following requirements.
+
+1. Binaries and metadata published for consumption by out-of-tree builds must be signed and automatically verified.
+2. Out-of-tree builds must support generating license attribution documents.
+3. All functionality necessary to build, test, and publish out-of-tree builds must be provided by a unified CLI.
+4. Maintainers must be able to define their own packages.
+5. Maintainers must be able to mix and match core project packages with their own.
+6. Maintainers must be able to replace a core project package with their own.
+7. Maintainers must be able to opt out of certain Bottlerocket subsystems by choosing not to install them.
+8. Maintainers must be able to use core project packages without building them.
```

Comment:
ie: host container runtime?

Reply:
> ie: host container runtime?

Yes, I will add a note so that we understand the purpose behind this requirement.